### PR TITLE
Readability change to `flutter.groovy`, align on null assignment, reduce unused scope for some methods, apply static where possible

### PR DESCRIPTION
**Pull request by AbdeMohlbi**
removed public modifier from this methods :`getVersionCode` , `getVersionName`.
add static to :`pluginSupportsAndroidPlatform` ,`buildGradleFile`,`settingsGradleFile`
`getCompileSdkFromProject` ,`getAssembleTask` 
refactor `==null` usage  to `:?` to unify the usage 

see https://github.com/flutter/flutter/issues/147122 for context

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by AbdeMohlbi** [comment]
windows tests fail , mac tests pass. 
mac tests pass , windows test fail 🤔

**Comment by stuartmorgan** [comment]
test-exempt: code refactor with no semantic change

**Comment by reidbaker** [comment]
@AbdeMohlbi I am going to give this a pass but can you update the title and pr description to give a more precise description of what is being refactored and why? It is helpful when reviewing all commits post release and also gives your code reviews more context.

File: packages/flutter_tools/gradle/src/main/groovy/flutter.groovy [hunk 1]
```diff
@@ -824,25 +818,17 @@ class FlutterPlugin implements Plugin<Project> {
...
-        int minNumTokens = Math.min(version1numTokens, version2numTokens)
-        for (int i = 0; i < minNumTokens; i++) {
-            int num1 = version1Tokenized[i].toInteger()
-            int num2 = version2Tokenized[i].toInteger()
-            if (num1 > num2) {
-                return version1
-            }
-            if (num2 > num1) {
-                return version2
+        List<Integer> version1Tokenized = version1.tokenize(".")*.toInteger()
```
Comment (reidbaker) [inline packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:821]: Semantically I don't think this is the same. Specifically I think a number like 8.6-rc-2 would crash on the first 2 lines.

**Comment by AbdeMohlbi** [comment]
> @AbdeMohlbi I am going to give this a pass but can you update the title and pr description to give a more precise description of what is being refactored and why? It is helpful when reviewing all commits post release and also gives your code reviews more context. 

No problem

**Review by reidbaker — CHANGES_REQUESTED** [review]
Unlike some of the other pr's this one has a mix of lint style fixes and some opinion style fixes. 
FWIW I think moving from `if (something == null)` set a default to using the ?: syntax is a bit more readable. 

Adding static where possible is a good change. 
reducing the scope of methods that dont need to be public is ok but I need to do a pass to make sure we dont actually depend on those methods someplace else.

**Comment by reidbaker** [comment]
Test-exemption: This pr should not land with any behavioral changes. If there are behavioral changes then tests need to be added.

**Comment by AbdeMohlbi** [comment]
> Unlike some of the other pr's this one has a mix of lint style fixes and some opinion style fixes. 
> FWIW I think moving from `if (something == null)` set a default to using the ?: syntax is a bit more readable. 
> 
> Adding static where possible is a good change. 
> reducing the scope of methods that dont need to be public is ok but I need to do a pass to make sure we dont actually depend on those methods someplace else. 

Well the file contains multiple instances of both , i tried unify the usage to ?:

**Comment by reidbaker** [comment]
> > Unlike some of the other pr's this one has a mix of lint style fixes and some opinion style fixes.
> > FWIW I think moving from `if (something == null)` set a default to using the ?: syntax is a bit more readable.
> > Adding static where possible is a good change.
> > reducing the scope of methods that dont need to be public is ok but I need to do a pass to make sure we dont actually depend on those methods someplace else.
> 
> Well the file contains multiple instances of both , i tried unify the usage to ?:

Yeah sorry I this was mostly a list of things I liked in the pr.

File: packages/flutter_tools/gradle/src/main/groovy/flutter.groovy (hunk 1 above)
Comment (AbdeMohlbi) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:821]: yep it should be `List<String>` but still this does not resolve cases like `8.6-rc-2` , like the TODO above says. 
hmmm , should i work for an overall fix for this or just let it be list<String> ?

Comment (reidbaker) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:821]: Most of this pr is good, lets remove this change and land the rest. Then you can either tackle this or not based on what you prefer.

Comment (AbdeMohlbi) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:821]: currenlty i'm seeing [a list of all gradle versions ](https://services.gradle.org/distributions/) (3160 as of today) ,i see that there is only the special case of `rc` , but i have no idea how to compare them for exemple is `8.7-rc-1` with `8.8` , i guess 8.8 is newer

Comment (AbdeMohlbi) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy]: well i found some script , tweaked it a bit and it seam to wrok :
(could u kindly confirm that this the intended behavior ?)
```groovy
def compareGradleVersions(String version1, String version2) {

    def v1Parts = version1.tokenize('.-')
    def v2Parts = version2.tokenize('.-')

    
    for (int i = 0; i < Math.max(v1Parts.size(), v2Parts.size()); i++) {
        
        def v1Part = i < v1Parts.size() ? v1Parts[i] : '0'
        def v2Part = i < v2Parts.size() ? v2Parts[i] : '0'

        
        def v1Num = v1Part.isNumber() ? v1Part.toInteger() : 0
        def v2Num = v2Part.isNumber() ? v2Part.toInteger() : 0

        if (v1Num != v2Num) {
            return v1Num > v2Num ? version1 : version2
        }

        // like -rc-1 cass
        if (v1Part.isNumber() && !v2Part.isNumber()) {
            return version1
        } else if (!v1Part.isNumber() && v2Part.isNumber()) {
            return version2
        } else if (v1Part != v2Part) {
            return comparePreReleaseIdentifiers(v1Part, v2Part) ? version1 : version2
        }
    }
    return version1 
    // If both are same return one of thm 
}

def comparePreReleaseIdentifiers(String v1Part, String v2Part) {
    // i tested "rc-1" vs "rc-2" worked (rc-2)
    // i tested "8.7-rc-1" vs "8.7" worked (8.7)
    // i tested "8.7-rc-1" vs "8.7.2" worked (8.7.2)
    // i tested "8.7.1" vs ""8.7.2"" worked (8.7.2)
    def v1PreRelease = v1Part.replaceAll("\\D", "")
    def v2PreRelease = v2Part.replaceAll("\\D", "")
    
    return v1PreRelease < v2PreRelease
}


def newerVersion = compareGradleVersions("8.7.1", "8.7.2")
println "The newer version is: $newerVersion"

```

**Review by reidbaker — APPROVED** [review]

**Comment by AbdeMohlbi** [comment]
related #158020

**Review by gmackall — APPROVED** [review]

**Pull request merged**